### PR TITLE
Redux hooks

### PR DESCRIPTION
**Pull request by shiranka**

### CODE REVIEW

File: my-app/src/components/TaskForm.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            id: Math.random(),
+            content: task
+        })
+        setTask('')
+    }    
+    return (
+        <div>
+            <form onSubmit={handleSubmit}>
+                <label>Add new task:</label>
+                <input type="text" onChange={handleChange} placeholder='Add a task...' value={task}></input>
```
Comment (yotamberk) [inline my-app/src/components/TaskForm.js]:
``` javascript
<input
   type="text" 
   placeholder='Add a task...' 
   onChange={handleChange} 
   value={task} 
/>
```

File: my-app/src/components/TaskForm.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const handleChange = (e) => { setTask(e.target.value) }
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        addTask({
+            id: Math.random(),
+            content: task
+        })
+        setTask('')
+    }    
+    return (
```
Comment (yotamberk) [inline my-app/src/components/TaskForm.js]: add a new line

File: my-app/src/components/TaskForm.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { addTaskAction } from '../redux'
+
+const TaskForm = () => {
+    const [task, setTask] = useState('')
+    const dispatch = useDispatch()
+    const addTask = (task) => dispatch(addTaskAction(task))
+    
+    const handleChange = (e) => { setTask(e.target.value) }
```
Comment (yotamberk) [inline my-app/src/components/TaskForm.js]: use `useCallback()` to remove the redefinition of this function.

File: my-app/src/components/TaskForm.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    const handleSubmit = (e) => {
```
Comment (yotamberk) [inline my-app/src/components/TaskForm.js]: same here. use `useCallback`

File: my-app/src/components/TasksList.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { deleteTaskAction} from '../redux'
+
+
+const TasksList = () => {
+    const tasks = useSelector((state) => state.tasks)
+    const dispatch = useDispatch()
+    const deleteTask = (id) => dispatch(deleteTaskAction(id))
```
Comment (yotamberk) [inline my-app/src/components/TasksList.js]: convert to be a closure: `(id) => () => dispatch(deleteTaskAction(id))` and wrap with `useCallback` so it gets memoized

File: my-app/src/components/TasksList.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    const tasks = useSelector((state) => state.tasks)
+    const dispatch = useDispatch()
+    const deleteTask = (id) => dispatch(deleteTaskAction(id))   
+    
+    const tasksList = (tasks.length) ? (
+        tasks.map(task => {
+            return (
+                <div className="collection-item" key={task.id}>
+                <span>{task.content}</span>
+                <button onClick={() => deleteTask(task.id)}>x</button>
```
Comment (yotamberk) [inline my-app/src/components/TasksList.js]: and then you can use `onClick={deleteTask(task.id)}` without redefining this function

File: my-app/src/redux.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+export const store = createStore(
+    reducer,
+    initialState
+)
+
+function reducer(state, {type, payload}){
+    switch(type){
+        case 'add_task':
```
Comment (yotamberk) [inline my-app/src/redux.js]: reducer types must be all capital like `ADD_TASK`

**Review by yotamberk — CHANGES_REQUESTED** [review]

**Review by yotamberk — APPROVED** [review]
Nice and clean! Very good!

File: my-app/src/components/TasksList.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    const dispatch = useDispatch()
+    
+    const deleteTask = useCallback((id) => () => {
+        dispatch(deleteTaskAction(id))
+    }, [dispatch])
+    
+    const tasksList = (tasks.length) ? (
+        tasks.map(task => {
+            return (
+                <div className="collection-item" key={task.id}>
```
Comment (avishag410) [inline my-app/src/components/TasksList.js]: there is no existing class "collection-item"

File: my-app/src/redux.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import { createStore } from 'redux'
```
Comment (avishag410) [inline my-app/src/redux.js]: please separate the file to actions and reducres

File: my-app/src/components/TaskForm.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        setTask('')
+    }, [setTask, task, dispatch])
+
+    return(
+        <div>
+            <form onSubmit={handleSubmit}>
+                <label>Add new task:</label>
+                <input
+                    type="text"
+                    onChange={handleChange}
```
Comment (avishag410) [inline my-app/src/components/TaskForm.js]: when calling on change from component, I'd rather u will use the following scheme:

<MyComponent
...
onChange={(e) => handleChange(e.target.value)}
/>

.....
const ... handleChange(Input) {
 setSomething(input)
}

File: my-app/src/components/TaskForm.js [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
-import { addTaskAction } from '../redux'
+import { addTaskAction } from '../redux/actions/task_action'
 
 const TaskForm = () => {
     const [task, setTask] = useState('')
     const dispatch = useDispatch()
    
-    const handleChange = useCallback((e) => {
-        setTask(e.target.value)
+    const handleChange = useCallback((input) => {
```
Comment (avishag410) [inline my-app/src/components/TaskForm.js]: your changes are what i meant, but use a different variable name, `input` is not meaningful

File: my-app/src/redux/store.js [hunk 12]
```diff
@@ -0,0 +1,14 @@
+import { createStore } from 'redux'
```
Comment (avishag410) [inline my-app/src/redux/store.js]: you don't need a different file for the store, it should be in the file with the reducer

**Pull request merged**